partiql-ast/src/main/java/org/partiql/ast/WindowFunctionType.java

```diff
@@ -0,0 +1,160 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.ast.expr.Expr;
+import org.partiql.ast.expr.ExprWindowFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a window function type.
+ * @see ExprWindowFunction#getFunctionType() 
+ */
+public abstract class WindowFunctionType extends AstNode {
+    /**
+     * A window function that takes no arguments.
+     * @see ExprWindowFunction#getFunctionType()
+     * @see WindowFunctionSimpleName
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static final class NoArg extends WindowFunctionType {
```

Comment:
Hmmm, mind evaluating why the choice of `NoArg` vs say `Ranking` and `RowNumber`. 

So seemingly for one we are modeling using number of args, for the other we are modeling using "function characteristic" for lack of better words. 

I understand that both `Ranking` and `RowNumber` must not contains argument... And no arg is, well, no arg... and you don't have to worry about argument type....

But perhaps using `NoArg` is bit over-smart, in that it might cause difficulty when (especially a user is) constructing the ast.